kong/tools/queue_schema.lua

```diff
@@ -0,0 +1,42 @@
+local Schema = require "kong.db.schema"
+
+-- TODO: enable the descriptions once they are accepted in Schemas
+return Schema.define {
+  type = "record",
+  fields = {
+    { name = {
+      type = "string",
+      -- description = "name of the queue",
+    } },
+    { batch_max_size = {
+      type = "number",
+      default = 1,
+      -- description = "maximum number of entries to be that can be processed at a time"
+    } },
+    { max_delay = {
+      type = "number",
+      default = 1,
+      -- description = "maximum number of (fractional) seconds to elapse after the first entry was queued before the queue starts calling the handler",
+    } },
+    { capacity = {
+      type = "number",
+      default = 10000,
+      -- description = "maximum number of entries that can be waiting on the queue",
```

Comment:
10.000 entries seems high enough to not make queues run of space in modest trafic situations.  If you have a better default and a rationale for that, please let me know.

> Does this need to have a max?

No.  If we chose too low, it would be inconvenient for big systems, if it is too high, then we can as well not have a max because before that is configured, the system runs out of memory.

Reply:
10k it is - what is the memory impact of buffering 10k requests today? Please quantify and add that to the code as a comment.